packages/flutter/lib/src/material/tabs.dart

```diff
@@ -562,8 +562,12 @@ class TabBarSelectionState<T> extends State<TabBarSelection<T>> with SingleTicke
   @override
   void didUpdateConfig(TabBarSelection<T> oldConfig) {
     super.didUpdateConfig(oldConfig);
+    _previousValue = _value;
+    _value = config.value ?? values.first;
```

Comment:
I don't understand why this line is needed at all. `_value` is the current state, it shouldn't change when the widget changes.